Tools/autotest/common.py

```diff
@@ -1677,31 +1657,26 @@ def test_arm_feature(self):
             if not self.disarm_motors_with_switch(arming_switch):
                 raise NotAchievedException("Failed to disarm with switch")
             self.set_rc(arming_switch, 1000)
-        if self.mav.mav_type in [mavutil.mavlink.MAV_TYPE_QUADROTOR,
-                                 mavutil.mavlink.MAV_TYPE_HELICOPTER,
-                                 mavutil.mavlink.MAV_TYPE_HEXAROTOR,
-                                 mavutil.mavlink.MAV_TYPE_OCTOROTOR,
-                                 mavutil.mavlink.MAV_TYPE_COAXIAL,
-                                 mavutil.mavlink.MAV_TYPE_TRICOPTER]:
-            self.start_test("Test arming failure with throttle too high")
-            self.set_rc(3, 1800)
-            try:
-                if self.arm_vehicle():
-                    raise NotAchievedException("Armed when throttle too high")
-            except AutoTestTimeoutException():
-                pass
-            except ValueError:
-                pass
-            if self.arm_motors_with_rc_input():
-                raise NotAchievedException(
-                    "Armed via RC when throttle too high")
-            if self.arm_motors_with_switch(arming_switch):
-                raise NotAchievedException("Armed via RC when switch too high")
-            self.set_throttle_zero()
-            self.set_rc(arming_switch, 1000)
 
-        # Sub doesn't have 'stick commands'
-        if self.mav.mav_type != mavutil.mavlink.MAV_TYPE_SUBMARINE:
+            if self.is_copter():
+                self.start_test("Test arming failure with throttle too high")
+                self.set_rc(3, 1800)
+                try:
+                    if self.arm_vehicle():
+                        raise NotAchievedException("Armed when throttle too high")
+                except AutoTestTimeoutException():
```

Comment:
I wonder if we can get rid of the `TimeoutException`?

Reply:
yep we can